File: Programs/Majority_Element_I.cpp
#include<bits/stdc++.h>
using namespace std;

/* LeetCode */
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> freq;
        for(int& num: nums)
            freq[num]++;
        
        int n= nums.size();
        for(int& num: nums) {
            if(freq[num] > n/2)
                return num;
        }

        return -1;
    }
};


/* CodingNinjas */
int findMajorityElement(int arr[], int n) {
	unordered_map<int, int> freq;

	for(int i= 0; i<n; i++)
		freq[arr[i]]++;
	
	for(int i= 0; i<n; i++) {
		if(freq[arr[i]] > n/2)
			return arr[i];
	}

	return -1;
}

File: Programs/Majority_Element_I.py
from collections import Counter


""" LeetCode """
class Solution:
    def majorityElement(self, nums: list[int]) -> int:
        freq= Counter(nums)

        n= len(nums)
        for key, value in freq.items():
            if value > n//2:
                return key
        
        return -1


""" CodingNinjas """
def findMajorityElement(arr, n):
	freq= Counter(arr)

	for key, value in freq.items():
		if value > n//2:
			return key
	
	return -1

